# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ugrep/makefile.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Well - the dietlibc hardcoding does not work for us. We want control over the
gcc / libc used anyway - and DESTDIR, so ... a tiny rewrite was needed ...

  - Rene Rebe <rene@exactcode.de>

--- ugrep/Makefile	2001-04-19 14:15:16.000000000 +0200
+++ ugrep-fixed/Makefile	2005-02-06 12:00:42.928299424 +0100
@@ -1,20 +1,23 @@
-LIB=/lib
+
+CC	= gcc -s -Os
+PREFIX	= usr
 
 ugrep:		ugrep.o regex.o uio.o
-		gcc -nostdlib -s -o ugrep uio.o regex.o ugrep.o $(LIB)/dietstart.o $(LIB)/libdietc.a
-		strip -R .note -R .comment ugrep
+		$(CC) -o ugrep uio.o regex.o ugrep.o
 		ln -fs ugrep uegrep
-udgrep:		ugrep.o regex.o
-		gcc -o udgrep ugrep.o regex.o
 ugrep.o:	ugrep.c
-		gcc -g -Wall -Os -c ugrep.c
+		$(CC) -Wall -c ugrep.c
 regex.o:	regex.c regex.h
-		gcc -g -Wall -Os -c regex.c
+		$(CC) -Wall -c regex.c
 uio.o:		uio.c uio.h
-		gcc -g -Wall -Os -c uio.c
+		$(CC) -Wall -c uio.c
 clean:	
 		rm -f ugrep uegrep udgrep *.o
 tarball:
 		cd .. && tar cvvIf ugrep.tar.bz2 ugrep
+install:	ugrep
+		cp -av ugrep $(DESTDIR)/$(PREFIX)/bin/
+		cp -av uegrep $(DESTDIR)/$(PREFIX)/bin/
+
 test:		ugrep
 		./testrun
